cpp/arcticdb/version/version_utils.hpp

```diff
@@ -230,3 +230,40 @@ void fix_stream_ids_of_index_keys(
     const StreamId &stream_id,
     const std::shared_ptr<VersionMapEntry> &entry);
 }
+
+inline arcticdb::proto::descriptors::SortedValue deduce_sorted(arcticdb::proto::descriptors::SortedValue existing_frame, arcticdb::proto::descriptors::SortedValue input_frame){
+    using namespace arcticdb::proto;
+    using SortingState = descriptors::SortedValue;
+    auto final_state = SortingState::UNSORTED;
+    switch(existing_frame){
+        case SortingState::UNKNOWN:
+            if(input_frame != SortingState::UNSORTED){
+                final_state = SortingState::UNKNOWN;
+            }else{
+                final_state = SortingState::UNSORTED;
+            }
+            break;
+        case SortingState::ASCENDING:
+            if(input_frame  == SortingState::UNKNOWN){
+                final_state = SortingState::UNKNOWN;
+            }else if (input_frame  != SortingState::ASCENDING){
+                final_state = SortingState::UNSORTED;
+            }else{
+                final_state = SortingState::ASCENDING;
+            }
+            break;
+        case SortingState::DESCENDING:
+            if(input_frame  == SortingState::UNKNOWN){
+                final_state = SortingState::UNKNOWN;
+            }else if (input_frame  != SortingState::DESCENDING){
+                final_state = SortingState::UNSORTED;
+            }else{
+                final_state = SortingState::DESCENDING;
```

Comment:
I guess this only matters to `append()`, but does it actually check the input is less than the existing index?

I.e. would it allow appending `[9, 8, 7]` to `[3, 2, 1]`?

Reply:
that's a good point

Reply:
The answer is no. We do not allow it, but not as part of all the sorting exception new work. It is something was already controlled in arcticDB. I consider this as resolved

arcticdb_ext.exceptions.InternalException: E_ASSERTION_FAILURE Can't append dataframe with start index 2019-01-01 00:00:00.0 to existing sequence ending at 2020-01-20 00:00:00.0